featuretools/primitives/utils.py

```diff
@@ -126,3 +126,54 @@ def load_primitive_from_file(filepath):
         raise RuntimeError("More than one primitive defined in file %s" % filepath)
 
     return primitives[0]
+
+
+def serialize_primitive(primitive):
+    """build a dictionary with the data necessary to construct the given primitive"""
+    args_dict = {name: val for name, val in primitive.get_arguments()}
+    cls = type(primitive)
+    return {
+        'type': cls.__name__,
+        'module': cls.__module__,
+        'arguments': args_dict,
+    }
+
+
+def deserialize_primitive(primitive_dict, class_cache=None):
+    """
+    Construct a primitive from the given dictionary (output from
+    serialize_primitive).
+
+    class_cache: A dictionary where the keys are tuples of (class_name, module_name)
+    and the keys are the corresponding to primitive classes.
+    """
+    class_name = primitive_dict['type']
+    module = primitive_dict['module']
+    cache_key = (class_name, module)
+    if class_cache and cache_key in class_cache:
+        cls = class_cache[cache_key]
+    else:
+        # Search all descendants of PrimitiveBase for the class.
+        cls = _find_class_in_descendants(class_name, module, PrimitiveBase)
+        if not cls:
+            raise RuntimeError('Primitive "%s" in module "%s" not found' %
+                               (class_name, module))
+        if class_cache:
+            class_cache[cache_key] = cls
```

Comment:
This only caches the primitive class found by `_find_class_in_descendants` but I think we can cache any primitive class found using `_find_class_in_descendants` once we look at all of it's subclasses.